### PR TITLE
[sdk/python]: simplify TransactionFactory flow

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: sdk/python/tests/nem/test_TransactionFactory.py [hunk 1]
```diff
@@ -38,19 +38,39 @@ def create_factory(self, type_parsing_rules=None):
...
-		#   type parsing rule that is getting created is for STRING 'Address'
-		# * same happens for `rental_fee`
-		#  * signer_public_key is handled differently in Transaction factory, via hint:
-		#    {'signer_public_key': PublicKey} which maps it to SDK type CryptoTypes.PublicKey
 		factory = self.create_factory({
-			'Address': lambda x: x + ' but amazing',
-			'Amount': lambda _: 654321,
-			PublicKey: lambda address: address + ' PUBLICKEY'
+			Address: lambda x: f'{x} but amazing',
+			Amount: lambda _: 654321,
```
Comment (gimre-xymcity) [inline sdk/python/tests/nem/test_TransactionFactory.py:72]: this now tests slightly different thing, but not sure if that is a problem

File: sdk/python/tests/test_RuleBasedTransactionFactory.py [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+
+	def test_pod_parser_can_handle_typed_value(self):
+		public_key = PublicKey('364F3694A022DB4DC59558944707C6679F6FD7E1A7B99CDE8F7D16D3FF515D28')
+		self._assert_pod_parser(public_key, public_key)
+
+	def test_pod_parser_uses_type_rule_override_when_available(self):
+		self._assert_pod_parser(
+			PublicKey('364F3694A022DB4DC59558944707C6679F6FD7E1A7B99CDE8F7D16D3FF515D28'),
+			'pubkey 364F3694A022DB4DC59558944707C6679F6FD7E1A7B99CDE8F7D16D3FF515D28',
+			{PublicKey: lambda value: f'pubkey {value}'})
```
Comment (gimre-xymcity) [inline sdk/python/tests/test_RuleBasedTransactionFactory.py:124]: one more test for string-based overrides?
(i.e. via 'Amount' and `class UnresolvedMosaic`)

**Review by gimre-xymcity — APPROVED** [review]

File: sdk/python/symbolchain/RuleBasedTransactionFactory.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+				continue
+
+			if issubclass(cls, BaseValue) and cls != BaseValue:
+				self.add_pod_parser(class_name, cls)
+
+			if issubclass(cls, Enum) and cls != Enum and cls != Flag:
+				if issubclass(cls, Flag):
+					self.add_flags_parser(class_name)
+				else:
+					self.add_enum_parser(class_name)
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply sdk/python/symbolchain/RuleBasedTransactionFactory.py:145]: i think it's ok. there shouldn't really be any spurious types in schemas.

File: sdk/python/tests/nem/test_TransactionFactory.py (hunk 1 above)
Comment (Jaguar0625) [reply sdk/python/tests/nem/test_TransactionFactory.py:72]: yea, but, i think it is ok

File: sdk/python/symbolchain/nem/TransactionFactory.py [hunk 4]
```diff
@@ -67,3 +56,37 @@ def attach_signature(transaction, signature):
...
+			'MosaicProperty', 'SizePrefixedMosaicProperty', 'MosaicDefinition',
+			'MultisigAccountModification', 'SizePrefixedMultisigAccountModification'
+		]
+		for name in struct_names:
+			factory.add_struct_parser(name)
+
+		sdk_type_mapping = {
+			'Address': Address,
+			'Hash256': Hash256,
+			'PublicKey': PublicKey,
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply sdk/python/symbolchain/nem/TransactionFactory.py:84]: well, we're using this for signier_pubkey field too, which is why i removed comment

File: sdk/python/tests/test_RuleBasedTransactionFactory.py (hunk 2 above)
Comment (Jaguar0625) [reply sdk/python/tests/test_RuleBasedTransactionFactory.py:124]: now it is only a type lookup, not a type name lookup too

File: sdk/python/symbolchain/nem/TransactionFactory.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (gimre-xymcity) [reply sdk/python/symbolchain/nem/TransactionFactory.py:84]: no, signer_pubkey is handled separately ^^ (this is for _OTHER_ fields)
```py
		processor.set_type_hints({'signer_public_key': PublicKey})

		transaction_type = processor.lookup_value('type')
		transaction = factory_class.create_by_name(transaction_type)
		transaction.signer_public_key = processor.lookup_value('signer_public_key')
```

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**